### PR TITLE
[move-vm] Cache verified modules

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description

Cache modules that have already been verified to avoid calling bytecode verifier.

### Test Plan
TBD

### CODE REVIEW

**Review by vgao1996 — APPROVED** [review]

File: third_party/move/move-vm/runtime/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,8 @@ edition = "2021"
 better_any = "0.1.1"
 bytes = "1.4.0"
 fail = "0.4.0"
+lazy_static = "1.4.0"
```
Comment (ziaptos) [inline third_party/move/move-vm/runtime/Cargo.toml:17]: These are already used in our codebase?

File: third_party/move/move-vm/runtime/src/loader/mod.rs [hunk 2]
```diff
@@ -84,6 +85,15 @@ where
     }
 }
 
+// Max number of modules that can skip re-verification.
+// 8 threads * (32 bytes + 2*16 for entries) * 100_000 < 80MB. Should be a reasonable memory consumption.
+const VERIFIED_CACHE_SIZE: usize = 100_000;
+
+// Cache for already verified modules
+lazy_static! {
+    static ref VERIFIED_MODULES: Mutex<lru::LruCache<[u8; 32], ()>> = Mutex::new(lru::LruCache::new(VERIFIED_CACHE_SIZE));
```
Comment (ziaptos) [inline third_party/move/move-vm/runtime/src/loader/mod.rs:94]: Why not a RWMutex?

**Review by ziaptos — APPROVED** [review]
We don't expect a lot of contention but would still be nice to replace the lock with a rwlock (as we expect a lot more reads than writes).

File: third_party/move/move-vm/runtime/Cargo.toml (hunk 1 above)
Comment (runtian-zhou) [reply third_party/move/move-vm/runtime/Cargo.toml:17]: Yes!

**Pull request merged**